kliff/dataset/weight.py

```diff
@@ -0,0 +1,167 @@
+from typing import Any, Dict, Optional
+import copy
+
+import numpy as np
+
+
+class Weight:
+    """
+    Base class for weight.
+
+    This class mimics the behavior  provided by the ``residual_data`` argument from the
+    previous version.
+    Subclass can implement
+        - compute_weight
+
+    Args:
+        config_weight: the weight for the entire configuration
+        energy_weight: the weight for the energy
+        forces_weight: the weight for the forces
+        stress_weight: the weight for the stress
+    """
+
+    def __init__(
+        self,
+        config_weight: float = 1.0,
+        energy_weight: float = 1.0,
+        forces_weight: float = 1.0,
+        stress_weight: float = 1.0,
+    ):
+        self._config_weight = config_weight  # weight for the entire configuration
+        self._energy_weight = energy_weight
+        self._forces_weight = forces_weight
+        self._stress_weight = stress_weight
+
+    @staticmethod
+    def compute_weight(config):
```

Comment:
Would be better to add typing for it. `(config:Configuration)`. Same for the NonuniformWeight below.

Reply:
What I noticed is that if I do `config:Configuration`, where is an importing loop, since `Configuration` is in dataset.py and that script also imports `Weight`.

Reply:
hmmm, let's skip the typing for now.

Reply:
Sounds good to me.